gtfu/src/main/java/gtfu/tools/TripListGenerator.java

```diff
@@ -177,6 +192,26 @@ private static Map<String, String> createCustomDirectionMap() {
         return directionMap;
     }
 
+    private static Map<String, List<String>> createServiceIDMap(int exception, CalendarDateCollection calendarDates){
+        Map<String, List<String>> dateMap = new HashMap();
+        CalendarDate currentCalDate = null;
+
+        for(int i = 0; i < calendarDates.getSize(); i++){
```

Comment:
CalendarDateCollection implements Iterable, so you can do:
`for (CalendarDate cd : calendarDates) {
...
}`